crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -1100,30 +1099,45 @@ impl CheckpointBuilder {
         let mut seen = HashSet::new();
         loop {
             let mut pending = HashSet::new();
+
+            let digests = roots
+                .iter()
+                .map(|e| *e.transaction_digest())
+                .collect::<Vec<_>>();
+
+            let included_transactions = self
+                .epoch_store
+                .builder_included_transactions_in_checkpoint(digests.iter())?;
+
+            let digest_inclusion_map: HashMap<_, _> = digests
+                .into_iter()
+                .zip(included_transactions.into_iter())
+                .collect();
```

Comment:
do you need a map here? the order and length of `included_transactions` should match the order and length of `roots`, so you should be able to zip them together and forego the map